### PR TITLE
[CARE-5694][CARE-1580] Feature - Implement improved version of [+] button

### PR DESCRIPTION
**Pull request by MohammadxAli**
## Mentions Dropdown (Removed `react-bootstrap` usage)

https://github.com/user-attachments/assets/d32bd5dd-d13c-4bd3-855f-389be25fd386

## New [+] menu

- Changed icons
- Improved styles
- Removed `react-bootstrap` usage

https://github.com/user-attachments/assets/0c522e0f-2003-4c41-a723-e3e64ac39572

### CODE REVIEW

File: babel.config.json [hunk 1]
```diff
@@ -4,19 +4,42 @@
...
+                                    "overrides": {
+                                        "removeViewBox": false,
+                                        "cleanupIds": false
+                                    }
+                                }
+                            }
+                        ]
+                    }
+                }
+            }
```
Comment (MohammadxAli) [inline babel.config.json:26]: This took me a while to understand how to properly deactivate cleaning up ids plugin of SVGO, if you're  interested to know why is this needed, please see: https://github.com/prezly/slate/pull/477#discussion_r1304336589

File: packages/slate-editor/.svgo.yml [hunk 2]
```diff
@@ -1,2 +0,0 @@
-plugins:
```
Comment (MohammadxAli) [inline packages/slate-editor/.svgo.yml:1]: This doesn't work, see my other comment.

File: packages/slate-editor/src/extensions/mentions/components/MentionsDropdown.tsx [hunk 3]
```diff
@@ -97,3 +96,21 @@ export const MentionsDropdown = <V extends object>({
         </ul>
     );
 };
+
+function MenuItem({
+    active,
+    children,
+    className,
+    ...props
+}: ComponentPropsWithRef<'li'> & { active?: boolean }) {
```
Comment (e1himself) [inline packages/slate-editor/src/extensions/mentions/components/MentionsDropdown.tsx:105]: > `ComponentPropsWithRef`

Hm... I wonder if this is necessary. `ref` is not a normal property, and can be assigned to any element without declaring it.

Moreover, you will never receive the `ref` value here, as a property. So you want a way to forward the ref of the `li` it will not work this way. You either need `forwardRef()`, or use a different property name (like `innerRef`, or something).

**Review by e1himself — APPROVED** [review]

File: packages/slate-editor/src/extensions/mentions/components/MentionsDropdown.tsx (hunk 3 above)
Comment (MohammadxAli) [reply packages/slate-editor/src/extensions/mentions/components/MentionsDropdown.tsx:105]: Ah good catch, there's also another type called [`ComponentPropsWithoutRef`](https://stackoverflow.com/a/73050903/6622233), I guess I'll need to use that.

Comment (e1himself) [reply packages/slate-editor/src/extensions/mentions/components/MentionsDropdown.tsx:105]: We usually use `HTMLAttributes` in such cases. Here I'd define it as `HTMLAttributes<HTMLLIElement>`.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,14 @@
+.PHONY: watch clean fresh
+
+clean:
```
Comment (kudlajz) [inline Makefile:3]: There's already a clean script in package.json, why do we have this one here and why is it different?

**Review by kudlajz — APPROVED** [review]

File: Makefile (hunk 4 above)
Comment (MohammadxAli) [reply Makefile:3]: It's not very different honestly, during linking/unlinking deleting `node_modules`, `.turbo` and `build` recursively is something I do a lot so I thought it's probably good to have it solved with a `make` task.

Basically it's the same but doesn't depend on `rimraf`. Should I remove it?

Comment (kudlajz) [reply Makefile:3]: I don't know 🙂 I've just installed `rimraf` globally and then it's not a problem to run `pnpm clean` whenever needed. So not sure if we need a Makefile that kinda does the same thing 🤔

Comment (MohammadxAli) [reply Makefile]: You're right :) Dropped it in favor of `npm` scripts.
